src/Microsoft.Azure.WebJobs.ServiceBus/Triggers/BrokeredMessageToStringConverter.cs

```diff
@@ -2,54 +2,76 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Globalization;
 using System.IO;
+using System.Runtime.Serialization;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Xml;
 using Microsoft.Azure.WebJobs.Host.Converters;
 using Microsoft.ServiceBus.Messaging;
 
 namespace Microsoft.Azure.WebJobs.ServiceBus.Triggers
 {
     internal class BrokeredMessageToStringConverter : IAsyncConverter<BrokeredMessage, string>
     {
-        public Task<string> ConvertAsync(BrokeredMessage input, CancellationToken cancellationToken)
+        public async Task<string> ConvertAsync(BrokeredMessage input, CancellationToken cancellationToken)
         {
             if (input == null)
             {
                 throw new ArgumentNullException("input");
             }
+            Stream stream = input.GetBody<Stream>();
+            if (stream == null)
+            {
+                return null;
+            }
 
-            if (input.ContentType == ContentTypes.TextPlain ||
```

Comment:
Unfortunately no -- there's really no failure for the byte[] converter so there's nothing to fallback to. Since we're just straight copying the bytes, it won't ever hit any Decoder exception (or similar) -- it doesn't care, it'll copy the bytes happily either way. So we need to use the ContentType as a hint as to how things were encoded.